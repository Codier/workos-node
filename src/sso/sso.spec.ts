import WorkOS from '../workos';

describe('SSO', () => {
  describe('getAuthorizeURL', () => {
    describe('with no custom api hostname', () => {
      it('generates an authorize url with the default api hostname', () => {
        const workos = new WorkOS('sk_test_Sz3IQjepeSWaI4cMS4ms4sMuU');

        const url = workos.sso.authorizeURL({
          domain: 'lyft.com',
          projectID: 'proj_123',
          redirectURI: 'example.com/sso/workos/callback',
        });

        expect(url).toMatchSnapshot();
      });
    });

    describe('with a custom api hostname', () => {
      it('generates an authorize url with the custom api hostname', () => {
        const workos = new WorkOS('sk_test_Sz3IQjepeSWaI4cMS4ms4sMuU', {
          apiHostname: 'api.workos.dev',
        });

        const url = workos.sso.authorizeURL({
          domain: 'lyft.com',
          projectID: 'proj_123',
          redirectURI: 'example.com/sso/workos/callback',
        });

        expect(url).toMatchSnapshot();
      });
    });

    describe('with state', () => {
      it('generates an authorize url the provided state', () => {
        const workos = new WorkOS('sk_test_Sz3IQjepeSWaI4cMS4ms4sMuU');

        const url = workos.sso.authorizeURL({
          domain: 'lyft.com',
          projectID: 'proj_123',
          redirectURI: 'example.com/sso/workos/callback',
          state: 'custom state',
        });

        expect(url).toMatchSnapshot();
      });
    });
  });
});
