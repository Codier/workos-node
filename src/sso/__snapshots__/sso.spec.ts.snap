// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SSO getAuthorizationURL with a custom api hostname generates an authorize url with the custom api hostname 1`] = `"https://api.workos.dev/sso/authorize?client_id=proj_123&domain=lyft.com&redirect_uri=example.com%2Fsso%2Fworkos%2Fcallback"`;

exports[`SSO getAuthorizationURL with no custom api hostname generates an authorize url with the default api hostname 1`] = `"https://api.workos.com/sso/authorize?client_id=proj_123&domain=lyft.com&redirect_uri=example.com%2Fsso%2Fworkos%2Fcallback"`;

exports[`SSO getAuthorizationURL with state generates an authorize url the provided state 1`] = `"https://api.workos.com/sso/authorize?client_id=proj_123&domain=lyft.com&redirect_uri=example.com%2Fsso%2Fworkos%2Fcallback&state=custom%20state"`;
